### PR TITLE
Updated poco with an option to build static libraries

### PR DESCRIPTION
**Pull request by ashkan2200**
This is to make the formula able to compile static libraries for poco

### CODE REVIEW

**Comment by ashkan2200** [comment]
Related to this issue https://github.com/Homebrew/homebrew/issues/47317

File: Library/Formula/poco.rb [hunk 1]
```diff
@@ -38,6 +39,11 @@ def install
     mkdir "macbuild" do
       system "cmake", buildpath, *args
       system "make", "install"
+      if build.with? "static"
+      	args << "-DPOCO_STATIC=ON"
```
Comment (DomT4) [inline Library/Formula/poco.rb]: Can this not be inlined to the prior block rather than enforcing doing the entire build again?

**Comment by ashkan2200** [comment]
I looked in the CMakeLists.txt for poco and there is an option POCO_STATIC which is "ON" or "OFF". So you can't make both shared and static at the same time. That is why I do it like this.

File: Library/Formula/poco.rb [hunk 2]
```diff
@@ -15,6 +15,7 @@ class Poco < Formula
 
   option :cxx11
   option :universal
+  option "with-static", "Build static libraries"
```
Comment (mikemcquaid) [inline Library/Formula/poco.rb]: `"Build static libraries (instead of shared)"`

File: Library/Formula/poco.rb [hunk 3]
```diff
@@ -38,6 +39,11 @@ def install
     mkdir "macbuild" do
       system "cmake", buildpath, *args
       system "make", "install"
+      if build.with? "static"
+        args << "-DPOCO_STATIC=ON"
+        system "cmake", buildpath, *args
```
Comment (mikemcquaid) [inline Library/Formula/poco.rb]: I think it's nicer to not rerun `cmake` and just allow static or shared.

Comment (ashkan2200) [inline Library/Formula/poco.rb]: I have updated the formula accordingly.

File: Library/Formula/poco.rb (hunk 2 above)
Comment (mikemcquaid) [inline Library/Formula/poco.rb]: @ashkan2200 Still need this bit updated.

Comment (ashkan2200) [inline Library/Formula/poco.rb]: @mikemcquaid Done

**Comment by mikemcquaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

For future reference the preferred commit message format is `poco: add static library option.`.

**Pull request closed without merging**